### PR TITLE
Add App Service security recommendations

### PR DESCRIPTION
**Pull request by jelledruyts**

### CODE REVIEW

File: checklists/appsvc_checklist.json [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+      "description": "Protect against malicious inbound traffic using a Web Application Firewall like Application Gateway or Azure Front Door.  Make sure to monitor the WAF's logs.",
+      "guid": "b123071a-5416-4415-a33e-a3ad2c2de732",
+      "severity": "High",
+      "link": "https://docs.microsoft.com/azure/app-service/networking/app-gateway-with-service-endpoints"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
+      "text": "Avoid for WAF to be bypassed",
+      "description": "Make sure the WAF cannot be bypassed by locking down access to only the WAF.  Use a combination of Access Restrictions, Service Endpoints and Private Endpoints.",
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:44]: Should an addition be made here for header and FDID check for AFD?

File: checklists/appsvc_checklist.json [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+      "description": "Set minimum TLS policy to 1.2 in App Service configuration.",
+      "guid": "c115775c-2ea5-45b4-9ad4-8408ee72734b",
+      "severity": "Medium",
+      "link": "https://docs.microsoft.com/azure/app-service/configure-ssl-bindings#enforce-tls-versions"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
+      "text": "Use HTTPS only",
+      "description": "Configure App Service to use HTTPS only.  This causes App Service to redirect from HTTP to HTTPS.  Strongly consider the use of HSTS in your code or from your WAF.",
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:62]: I have no idea what HSTS is ... could we add short explanation? Linked docs link also doesn't mention HSTS. 

Google helps out though: HTTP Strict Transport Security (HSTS)

File: checklists/appsvc_checklist.json [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+      "subcategory": "Network Security",
+      "text": "Wildcards must not be used for CORS",
+      "description": "Do not use wildcards in your CORS configuration, as this allows all origins to access the service (thereby defeating the purpose of CORS). Specifically only allow the origins that you expect to be able to access the service.",
+      "guid": "68266abc-a264-4f9a-89ae-d9c55d04c2c3",
+      "severity": "High",
+      "link": "https://docs.microsoft.com/azure/app-service/app-service-web-tutorial-rest-api"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:78]: isn't this rather identity security? (or another subcategory, seems unrelated to network security)

File: checklists/appsvc_checklist.json [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+      "text": "Use HTTPS only",
+      "description": "Configure App Service to use HTTPS only.  This causes App Service to redirect from HTTP to HTTPS.  Strongly consider the use of HSTS in your code or from your WAF.",
+      "guid": "475ba18f-dbf5-490c-b65d-e8e03f9bcbd4",
+      "severity": "High",
+      "link": "https://docs.microsoft.com/azure/app-service/configure-ssl-bindings#enforce-https"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:69]: Not really network security this.

File: checklists/appsvc_checklist.json [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+      "text": "Use an established Identity Provider for authentication",
+      "description": "For authenticated web application, use a well established Identity Provider like Azure AD or Azure AD B2C.  Leverage the application framework of your choice to integrate with this provider or use the App Service Authentication / Authorization feature.",
+      "guid": "919ca0b2-c121-459e-814b-933df574eccc",
+      "severity": "Medium",
+      "link": "https://docs.microsoft.com/azure/app-service/overview-authentication-authorization"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:87]: idem

File: checklists/appsvc_checklist.json [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+      "text": "Turn off remote debugging",
+      "description": "Remote debugging must not be turned on in production as this opens additional ports on the service which increases the attack surface. Note that the service does turn of remote debugging automatically after 48 hours.",
+      "guid": "d9bd3baf-cda3-4b54-bb2e-b03dd9a25827",
+      "severity": "High",
+      "link": "https://docs.microsoft.com/azure/app-service/configure-common#configure-general-settings"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:96]: idem

File: checklists/appsvc_checklist.json [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+      "text": "Ensure a stable IP for outbound communications towards internet adresses",
+      "description": "You can provide a stable outbound IP by using VNet integration and using a VNet NAT Gateway or an NVA like Azure Firewall.  This allows the receiving party to allow-list based on IP, should that be needed.  Note that for communications towards Azure Services often there's no need to depend on the IP address and mechanics like Service Endpoints should be used instead.  (Also the use of private endpoints on the receiving end avoids for SNAT to happen and provides a stable outbound IP range.)",
+      "guid": "cda3b54e-b2eb-403d-b9a2-582718d2ddb1",
+      "severity": "Low",
+      "link": "https://docs.microsoft.com/azure/app-service/networking/nat-gateway-integration"
+    },
+    {
+      "category": "Security",
+      "subcategory": "Network Security",
+      "text": "Inbound network access should be controlled",
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:25]: Maybe add a similar check for the SCM site

File: checklists/appsvc_checklist.json [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+    {
+      "category": "Security",
+      "subcategory": "Penetration Testing",
+      "text": "Conduct a penetration test",
+      "description": "Conduct a penetration test on the web application following the penetration testing rules of engagement.",
+      "guid": "eb2eb03d-d9a2-4582-918d-2ddb10725769",
+      "severity": "Medium",
+      "link": "https://docs.microsoft.com/azure/security/fundamentals/pen-testing"
+    }
+  ],
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:247]: Maybe want to add: Use ASE in case you need isolated dedicated compute environment

File: checklists/appsvc_checklist.json [hunk 9]
```diff
@@ -0,0 +1,304 @@
...
+  "categories": [
+    {
+      "name": "Identity and Access Management"
+    },
+    {
+      "name": "Network Topology and Connectivity"
+    },
+    {
+      "name": "Business Continuity and Disaster Recovery"
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:256]: Do we want to add additional items for this subcategory? Would make sense to add some stuff here maybe?

File: checklists/appsvc_checklist.json [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+    },
+    {
+      "name": "Security, Governance, and Compliance"
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:259]: Maybe here as well, something on the policies for App Service and/or to set up Azure RBAC.

File: checklists/appsvc_checklist.json [hunk 11]
```diff
@@ -0,0 +1,304 @@
...
+    },
+    {
+      "name": "Application Automation and DevOps"
```
Comment (vermegi) [inline checklists/appsvc_checklist.json:262]: Maybe add use of staging environments for zero downtime deploy?

File: checklists/appsvc_checklist.json (hunk 1 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:44]: Hmm, it's mentioned in the linked doc and I wouldn't add too much detail in the checklist itself.

File: checklists/appsvc_checklist.json (hunk 2 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:62]: Added!

File: checklists/appsvc_checklist.json (hunk 3 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:78]: Changed to "Identity and Access Control"

File: checklists/appsvc_checklist.json (hunk 4 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:69]: Why not? CORS is restricting the browser to make network requests unless explicitly allowed?

File: checklists/appsvc_checklist.json (hunk 5 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:87]: Why not? This disables the remote debugging port at the network level.

File: checklists/appsvc_checklist.json (hunk 6 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:96]: Defender "mostly" works at the networking layer, not sure where else to better put this?

File: checklists/appsvc_checklist.json (hunk 7 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:25]: Added a note calling this out in the same item.

File: checklists/appsvc_checklist.json (hunk 8 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:247]: That is called out (not in detail) in the "Isolate systems that process sensitive information" item.

File: checklists/appsvc_checklist.json (hunk 9 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:256]: This exercise is purely focused on security recommendations for now, so not at this time - but definitely feel free to add more items in those other categories. I'd argue it should then be fairly "exhaustive" though, meaning if you would tackle the BCDR category it shouldn't be a few random thoughts but ideally a relatively "complete" set of checklist items.

File: checklists/appsvc_checklist.json (hunk 10 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:259]: (As before)

File: checklists/appsvc_checklist.json (hunk 11 above)
Comment (jelledruyts) [reply checklists/appsvc_checklist.json:262]: (As before)

**Comment by erjosito** [comment]
@jelledruyts  Could you please rename to appsvc_checklist.en.json? (otherwise automatic translation will not work)
